gcp/table_gcp_project.go

```diff
@@ -0,0 +1,121 @@
+
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableGcpProject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "gcp_project",
+		Description: "GCP Project",
+		List: &plugin.ListConfig{
+			Hydrate: listGCPProjects,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "project_id",
+				Description: "The unique, user-assigned ID of the Project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "project_number",
+				Description: "The number uniquely identifying the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "Specifies the project lifecycle state.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "create_time",
+				Description: "Creation time of the project.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "parent",
+				Description: "An optional reference to a parent Resource.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "labels",
+				Description: "A list of labels attached to this project.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getProjectAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getProject,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listGCPProjects(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Service Connection
+	service, err := CloudResourceManagerService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+	resp, err := service.Projects.List().Filter("name=" + project).Do()
+	for _, project := range resp.Projects {
+		d.StreamListItem(ctx, project)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func getProjectAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
```

Comment:
Do we need this block?
As we are getting the project in response.

Reply:
This is required  to filter out the current project in list call

Reply:
@anisadas We are filtering project in GET call.
Why we need to filter when building `akas`?

Reply:
My bad I think you are talking this  block in list call . I understood it. Will change it